app/assets/javascripts/media_player_wrapper/mejs4_helper_markers.es6

```diff
@@ -359,26 +344,128 @@ class MEJSMarkersHelper {
   }
 
   /**
-   * Update markers in Mediaelement player's time rail by hooking into the Mediaelement Markers plugin
+   * Re-build the markers table and player markers
+   * @function rebuildMarkers
+   * @return {void}
+   */
+  rebuildMarkers() {
+    this.rebuildMarkersTable();
+    this.updateVisualMarkers();
+  }
+
+  /**
+   * Update markers in Mediaelement player's time rail
    * @function updateVisualMarkers
-   * @param {Array} markers Array of marker start times
    * @return {void}
    */
-  updateVisualMarkers(markers) {
-    const t = this;
-    const player = mejs4AvalonPlayer.player;
-    player.options.markers = markers;
+  updateVisualMarkers() {
+    const playlistIds = this.getCurrentPlaylistIds();
 
-    // Directly delete current markers from the player UI
-    let currentMarkerEls = player.controls.getElementsByClassName(
-      player.options.classPrefix + 'time-marker'
+    this.getMarkers(playlistIds.playlistId, playlistIds.playlistItemId).then(
+      response => {
+        const markers = response;
+        const duration = mejs4AvalonPlayer.player.duration;
+        const scrubber = $('.mejs__time-rail').css('position', 'relative');
```

Comment:
Don't change this now, but in the future if its a jQuery variable reference, call it `$scrubber` instead of `scrubber` so we can keep track of stuff at quick glances.